File: traffic/src/main/java/com/hsryuuu/traffic/tmp/coupon/CouponController.java
package com.hsryuuu.traffic.tmp.coupon;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/coupons")
public class CouponController {

    @PostMapping
    public ResponseEntity<Void> createCouponEvent(@RequestBody CreateCouponEventRequest request) {
        // TODO: 쿠폰 이벤트 생성 (총 수량, 기간 설정)
        return ResponseEntity.ok().build();
    }

    @PostMapping("/{couponEventId}/issue")
    public ResponseEntity<Void> issueCoupon(@PathVariable Long couponEventId,
                                            @RequestBody IssueCouponRequest request) {
        // TODO: 쿠폰 발급 요청 (선착순, 1인 1매)
        return ResponseEntity.ok().build();
    }

    @GetMapping("/{couponEventId}/status")
    public ResponseEntity<Void> getCouponEventStatus(@PathVariable Long couponEventId) {
        // TODO: 쿠폰 이벤트 현황 조회 (발급 수량, 잔여 수량)
        return ResponseEntity.ok().build();
    }

    @GetMapping("/users/{userId}")
    public ResponseEntity<Void> getUserCoupons(@PathVariable Long userId) {
        // TODO: 사용자 보유 쿠폰 목록 조회
        return ResponseEntity.ok().build();
    }

    record CreateCouponEventRequest(String title, int totalQuantity, String startAt, String endAt) {}
    record IssueCouponRequest(Long userId) {}
}

File: high-traffic/src/main/kotlin/com/hsryuuu/hightraffic/coupon/controller/CouponController.kt
package com.hsryuuu.hightraffic.coupon.controller

import com.hsryuuu.hightraffic.coupon.dto.CouponStatusResponse
import com.hsryuuu.hightraffic.coupon.dto.IssueCouponRequest
import com.hsryuuu.hightraffic.coupon.dto.IssueCouponResponse
import com.hsryuuu.hightraffic.coupon.dto.IssueCountResponse
import com.hsryuuu.hightraffic.coupon.service.CouponIssueService
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/coupons")
class CouponController(
    private val couponIssueService: CouponIssueService,
) {

    @PostMapping("/{couponId}/issue")
    fun issue(
        @PathVariable couponId: Long,
        @Valid @RequestBody request: IssueCouponRequest,
    ): IssueCouponResponse {
        couponIssueService.issueCoupon(couponId, request.userId)
        return IssueCouponResponse(couponId = couponId, userId = request.userId)
    }

    @GetMapping("/{couponId}")
    fun getCoupon(@PathVariable couponId: Long): CouponStatusResponse =
        couponIssueService.getCouponStatus(couponId)

    @GetMapping("/{couponId}/issues/count")
    fun getIssuedCount(@PathVariable couponId: Long): IssueCountResponse =
        IssueCountResponse(couponId, couponIssueService.countIssues(couponId))
}
